jmri_api.md

```diff
@@ -0,0 +1,4 @@
+Type | Name | Status | Force
+---- | ---- | ------ | -----
+package | jmri.** | EXPERIMENTAL | X
+package | jmri | MAINTAINED
```

Comment:
I strongly believe that the classes, interfaces, and enums in the jmri package should EXPERIMENTAL and not MAINTAINED unless deprecated, in which case they must be DEPRECATED.

Reply:
> I strongly believe that the classes, interfaces, and enums in the jmri package should EXPERIMENTAL and not MAINTAINED unless deprecated, in which case they must be DEPRECATED.

!!!!

The basic structure of `jmri.Turnout` is over 15 years old.  Under what circumstances would it "be removed without prior notice, even in a patch"?

To put it another way, if everything in `jmri` is EXPERIMENTAL right now, what isn't?

Reply:
When a bulk change is made and you can't tell what changed, everything should be the same, and then we discuss the repercussions of changing single classes, methods, or fields. #8789 had very specific field-level changes that mark 15 year old public fields in Turnout as INTERNAL details despite multiple non-Java programs depending on them, but the GitHub web interface would not let me comment on that because the PR was simply too big.

That's why I firmly believe that any bulk change must be a single change across the board (i.e. make *everything* EXPERIMENTAL or MAINTAINED or whatever), and that any deviations from that should be in smaller follow on PRs that allow for a discussion of a single part of the API.

Reply:
Sorry, I wasn't clear enough, but #8789 was an EXAMPLE. It was never intended to be the real thing. I created it to demonstrate how the table and the script works. My intention with that file was to get a table that tested different cases, for example having different API status on methods and fields in a class or interface. I now realized that I gave the PR a really bad name.

Reply:
If we want, we can create three different tables:
* jmri.jmrit
* jmri.jmrix
* jmri and the rest of the packages below jmri.

That would solve the problem with the too long PR.